src/System.Drawing.Common/src/System/Drawing/Drawing2D/PathGradientBrush.cs

```diff
@@ -13,47 +13,36 @@ public sealed class PathGradientBrush : Brush
     {
         public PathGradientBrush(PointF[] points) : this(points, WrapMode.Clamp) { }
 
-        public PathGradientBrush(PointF[] points, WrapMode wrapMode)
+        public unsafe PathGradientBrush(PointF[] points, WrapMode wrapMode)
         {
             if (points == null)
             {
                 throw new ArgumentNullException(nameof(points));
             }
-            
+
             if (wrapMode < WrapMode.Tile || wrapMode > WrapMode.Clamp)
             {
                 throw new InvalidEnumArgumentException(nameof(wrapMode), unchecked((int)wrapMode), typeof(WrapMode));
             }
 
-            IntPtr pointsBuf = SafeNativeMethods.Gdip.ConvertPointToMemory(points);
-            try
-            {
-                IntPtr nativeBrush;
-                int status = SafeNativeMethods.Gdip.GdipCreatePathGradient(new HandleRef(null, pointsBuf),
-                                                            points.Length,
-                                                            unchecked((int)wrapMode),
-                                                            out nativeBrush);
-
-                if (status != SafeNativeMethods.Gdip.Ok)
-                    throw SafeNativeMethods.Gdip.StatusException(status);
+            // GdipCreatePathGradient returns InsufficientBuffer for less than 3 points, which we turn into
+            // OutOfMemoryException(). We used to copy nothing into an empty native buffer for zero points,
+            // which gives a valid pointer. Fixing an empty array gives a null pointer, which causes an
+            // InvalidParameter result, which we turn into an ArgumentException. Matching the old throw.
+            if (points.Length == 0)
+                throw new OutOfMemoryException();
 
-                SetNativeBrushInternal(nativeBrush);
-            }
-            finally
+            fixed(PointF* p = points)
             {
-                // Inside SafeNativeMethods.Gdip.ConvertPointToMemory, Marshal.AllocHGlobal
-                // is used to allocate unmanaged memory. Therefore, we need to free it
-                // manually with Marshal.FreeHGlobal
-                if (pointsBuf != IntPtr.Zero)
-                {
-                    Marshal.FreeHGlobal(pointsBuf);
-                }
+                SafeNativeMethods.Gdip.CheckStatus(SafeNativeMethods.Gdip.GdipCreatePathGradient(
```

Comment:
Consider adding "using SafeNativeMethods". BTW, could you please point me to the .net core style guide?

Reply:
https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/coding-style.md

Reply:
> Consider adding "using SafeNativeMethods"

Yeah, I intend to do that in a separate change. Discussed with @safern offline. :)